scapy/base_classes.py

```diff
@@ -13,14 +13,16 @@
 
 from __future__ import absolute_import
 import re,random,socket
-import types
+import types, operator
+from functools import reduce
 from scapy.modules.six.moves import range
 
 class Gen(object):
     __slots__ = []
     def __iter__(self):
         return iter([])
-
+    def __iterlen__(self):
+        return len([x for x in iter(self)])
```

Comment:
You're creating an unnecessary list here. `sum(1 for _ in iter(self))` should do the thing as your current code, without creating a list.

Reply:
Nice catch !